LabExT/View/Movement/CoordinatePairingsWindow.py

```diff
@@ -0,0 +1,271 @@
+            self._device_table.pack(side=TOP, fill=X, expand=True)
+
+            self._select_device_button = Button(
+                frame,
+                text="Select marked device",
+                command=self._on_device_selection
+            )
+            self._select_device_button.pack(side=LEFT, pady=2)
+        else:
+            Label(
+                frame,
+                text=self._device.short_str(),
+                font='Helvetica 12 bold'
+            ).pack(side=LEFT, fill=X)
+
+            self._clear_device_button = Button(
+                frame,
+                text="Clear selection",
+                command=self._on_device_selection_clear
+            )
+            self._clear_device_button.pack(side=LEFT, padx=5)
+
+    #
+    #   Callbacks
+    #
+
+    def _finish(self):
+        try:
+            if self._in_calibration:
+                self._in_stage_coordinate = self._in_calibration.stage.get_current_position()
+            if self._out_calibration:
+                self._out_stage_coordinate = self._out_calibration.stage.get_current_position()
+
+            self.destroy()
+        except StageError as e:
+            messagebox.showerror(
+                "Error", "Could not get current position: {}".format(e))
+
+    def _cancel(self):
+        self._in_stage_coordinate = None
+        self._out_stage_coordinate = None
+        self._device = None
+        self.destroy()
+
+    def _on_device_selection(self):
+        """
+        Callback, when user hits "Select marked device" button.
+        """
+        self._device = self._device_table.get_selected_device()
+        if self._device is None:
+            messagebox.showwarning(
+                'Selection Needed',
+                'Please select one device.',
+                parent=self)
+            return
+
+        self.__reload__()
+
+    def _on_device_selection_clear(self):
+        """
+        Callback, when user wants to clear the current device selection.
+        """
+        self._device = None
+        self.__reload__()
+
+    #
+    #   Helpers
+    #
+
+    def _calibration_pairing_widget(self, parent, calibration):
+        pairing_frame = CustomFrame(parent)
+        pairing_frame.title = calibration.short_str
+        pairing_frame.pack(side=TOP, fill=X, pady=5)
+
+        Label(
+            pairing_frame,
+            text="Paired with chip coordinate:"
+        ).pack(side=LEFT, fill=X)
+
+        if self._device:
+            Label(
+                pairing_frame,
+                text=self._device._in_position if calibration.is_input_stage else self._device._out_position,
+                borderwidth=2,
+                relief='sunken').pack(
+                side=LEFT,
+                padx=5)
+        else:
+            Label(
+                pairing_frame,
+                text="No device selected",
+                foreground="#FF3333"
+            ).pack(side=LEFT, padx=5)
+
```

Comment:
A cool feature here would be to have a display of the current stage coordinates.

E.g. a "will be paired with stage coordinates: XXX YYY ZZZ" label or so.

This suggestion needs a background-running routine which reads and updates the coordinates from the stage accordingly.

Lets talk about it in the next meeting, don't spend time implementing it just yet :)

Reply:
In fact, I have such a feature in a private branch, so it's easy for me to add it. 😄 Will be up in a minute.

Reply:
<img width="1000" alt="Screenshot 2022-02-22 at 17 15 18" src="https://user-images.githubusercontent.com/14354617/155173213-0aa731e0-169f-4885-9ebc-d0fbf696eb7a.png">